x/posts/legacy/v0.15.0/migrate_test.go

```diff
@@ -0,0 +1,200 @@
+		RegisteredReactions: []v040posts.Reaction{
+			{
+				ShortCode: ":my_house:",
+				Value:     "https://myHouse.png",
+				Subspace:  subspace,
+				Creator:   postCreator,
+			},
+		},
+		Params: v080posts.Params{
+			MaxPostMessageLength:            sdk.NewInt(10),
+			MaxOptionalDataFieldsNumber:     sdk.NewInt(10),
+			MaxOptionalDataFieldValueLength: sdk.NewInt(10),
+		},
+	}
+
+	expectedGenState := v0150posts.GenesisState{
+		Posts: []v0150posts.Post{
+			{
+				PostID:         string(parentID),
+				ParentID:       "",
+				Message:        "Message",
+				AllowsComments: true,
+				Subspace:       "4e188d9c17150037d5199bbdb91ae1eb2a78a15aca04cb35530cccb81494b36e",
+				OptionalData:   []v0130.OptionalDataEntry{{Key: "optional", Value: "data"}},
+				Created:        parentCreationTime,
+				LastEdited:     time.Time{},
+				Creator:        parentPostCreator.String(),
+				Attachments:    []v0100posts.Attachment{{URI: "https://uri.com", MimeType: "text/plain", Tags: nil}},
+			},
+			{
+				PostID:         string(postID),
+				ParentID:       string(parentID),
+				Message:        "Message",
+				AllowsComments: true,
+				Subspace:       "4e188d9c17150037d5199bbdb91ae1eb2a78a15aca04cb35530cccb81494b36e",
+				OptionalData:   []v0130.OptionalDataEntry{{Key: "optional", Value: "data"}},
+				Created:        postCreationTime,
+				LastEdited:     time.Time{},
+				Creator:        postCreator.String(),
+				Attachments:    []v0100posts.Attachment{{URI: "https://uri.com", MimeType: "text/plain", Tags: nil}},
+			},
+		},
+		UsersPollAnswers: []v0150posts.UserAnswersEntry{
+			{
+				PostID: string(postID),
+				UserAnswers: []v0150posts.UserAnswer{
+					{
+						User:    postCreator.String(),
+						Answers: []string{"1", "2"},
+					},
+				},
+			},
+		},
+		PostReactions: []v0150posts.PostReactionsEntry{
+			{
+				PostID: string(postID),
+				Reactions: []v0150posts.PostReaction{
+					{
+						ShortCode: ":fire:",
+						Value:     "🔥",
+						Owner:     postCreator.String(),
+					},
+					{
+						ShortCode: ":my_house:",
+						Value:     "https://myHouse.png",
+						Owner:     postCreator.String(),
+					},
+				},
+			},
+		},
+		RegisteredReactions: []v0150posts.RegisteredReaction{
+			{
+				ShortCode: ":my_house:",
+				Value:     "https://myHouse.png",
+				Subspace:  subspace,
+				Creator:   postCreator.String(),
+			},
+		},
+		Params: v080posts.Params{
+			MaxPostMessageLength:            sdk.NewInt(10),
+			MaxOptionalDataFieldsNumber:     sdk.NewInt(10),
+			MaxOptionalDataFieldValueLength: sdk.NewInt(10),
+		},
+	}
+
+	migrated := v0150posts.Migrate(v0130genesisState)
+
+	// Check for posts
+	require.Len(t, expectedGenState.Posts, len(migrated.Posts))
+	for index, post := range migrated.Posts {
+		require.Equal(t, expectedGenState.Posts[index].PostID, post.PostID)
+		require.Equal(t, expectedGenState.Posts[index].ParentID, post.ParentID)
+		require.Equal(t, expectedGenState.Posts[index].OptionalData, post.OptionalData)
```

Comment:
In order to make sure we are migrating all the data, we should check all the different fields for equality. Also, it might be better to use a variable to simplify this check: 

```golang
		expected := expectedGenState.Posts[index]
		require.Equal(t, expected.PostID, post.PostID)
		require.Equal(t, expected.ParentID, post.ParentID)
		require.Equal(t, expected.OptionalData, post.OptionalData)
		require.Equal(t, expected.AllowsComments, post.AllowsComments)
		// ...
```